

This is a horrible bodge; but makes my life easier for 1.

Index: svtools/source/filepicker/inettbc.cxx
===================================================================
RCS file: /cvs/oo/util/svtools/source/filepicker/inettbc.cxx,v
retrieving revision 1.20
diff -u -p -u -r1.20 inettbc.cxx
--- svtools/source/filepicker/inettbc.cxx	10 Dec 2001 16:00:29 -0000	1.20
+++ svtools/source/filepicker/inettbc.cxx	25 Apr 2003 16:58:18 -0000
@@ -919,6 +919,31 @@ BOOL SvtURLBox::ProcessKey( const KeyCod
 	}
 
 	KeyCode aCode( rKey.GetCode() );
+
+#ifdef UNX
+	if ( aCode == KEY_TAB && !rKey.IsMod1() && !rKey.IsShift() ) {
+		String aCurText = GetText();
+		Selection aSelection( GetSelection() );
+
+		if( aSelection.Max() != aCurText.Len() )
+			return TRUE; // nothing to complete yet
+
+		if( !pImp->pCompletions || pImp->pCompletions->Count() > 1)
+			return TRUE;
+
+		aSelection.Max() = SELECTION_MAX;
+		aSelection.Min() = SELECTION_MAX;
+		SetSelection( aSelection );
+
+		// for file dialog
+		bAutoCompleteMode = FALSE;
+		GetOpenHdl().Call(this);
+
+		SetText( String() );
+		return TRUE;
+	}
+#endif
+
 	if ( aCode == KEY_RETURN && GetText().Len() )
 	{
         // wait for completion of matching thread
@@ -1045,7 +1070,9 @@ long SvtURLBox::Notify( NotifyEvent &rEv
 {
 	if ( EVENT_GETFOCUS == rEvt.GetType() )
     {
-#ifndef UNX
+#ifdef UNX
+	    SetSelection( Selection( SELECTION_MAX, SELECTION_MAX ) );
+#else
 		// pb: don't select automatically on unix #93251#
 		SetSelection( Selection( 0, GetText().Len() ) );
 #endif
